asg/src/expression/array_inline.rs

```diff
@@ -118,6 +119,27 @@ impl FromAst<leo_ast::ArrayInlineExpression> for ArrayInlineExpression {
         };
 
         let mut len = 0;
+
+        if len == 0 && expected_item.is_none() {
+            for expr in value.elements.iter() {
```

Comment:
this reprocessed the entire subnode here, which is a nontrivial perf cost in some cases and unnecessary